src/black/__init__.py

```diff
@@ -626,9 +645,7 @@ def get_sources(
                     gitignore,
                 )
             )
-        elif s == "-":
-            sources.add(p)
-        elif p.is_file():
+        elif p.is_file() or is_stdin:
```

Comment:
```suggestion
        elif is_stdin or p.is_file():
```

As per Jelle's good point about reducing disk reads.